lib/puppet-lint/plugins/check_unsafe_interpolations.rb

```diff
@@ -0,0 +1,28 @@
+PuppetLint.new_check(:check_unsafe_interpolations) do
+  COMMANDS = Array['command', 'onlyif', 'unless']
+  def check
+    # Gather any exec commands' resources into an array
+    exec_resources = resource_indexes.map do |resource|
+      resource if resource[:type].value == 'exec' && !(COMMANDS & resource[:param_tokens].map(&:value)).empty?
+    end
+
+    # Iterate over each command found in any exec
+    exec_resources.each do |command_resources|
+      # Filter each resource in command and check for unsaafe interpolations
+      command_resources[:tokens].select do |token|
```

Comment:
I'm not 100% sure select is the right method to use here because it essentially returns an array of items that match the condition in the code block.